api/src/main/java/org/apache/iceberg/RowIdentifyField.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.io.Serializable;
+import java.util.Objects;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.types.Types;
+
+/**
+ * A field in {@link RowIdentifier}
+ * <p>
+ * The field must be:
+ *  1. a required column in the table schema
+ *  2. a primitive type column
+ */
+public class RowIdentifyField implements Serializable {
+
+  private final int sourceId;
+
+  RowIdentifyField(int sourceId) {
+    this.sourceId = sourceId;
+  }
+
+  RowIdentifyField(Types.NestedField column) {
+    ValidationException.check(column.isRequired(),
+        "Cannot add column %s to row identifier because it is not a required column", column);
```

Comment:
Nit:  In this `RowIdentifyField` constructor,  we will throw the exception with message saying `Cannot add column ...` , that seems unreasonable because we are mixed the column adding and instance constructing together.   How about moving those validation check out of this constructor ?

Reply:
makes sense, will update